plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -38,6 +43,50 @@
 		ResourceKind.ImageStream, 
 		ResourceKind.Route
 	};
+	
+	private Map<IProject, List<ResourceGrouping>> groupMap = new HashMap<IProject, List<ResourceGrouping>>();
```

Comment:
wondering if we shouldnat have ResourceGrouping holding a reference to the project. IMHO this would be far cleaner than having mappings that we need to keep in sync?